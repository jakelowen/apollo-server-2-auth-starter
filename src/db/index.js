const environment = process.env.NODE_ENV || 'development';
const knexTinyLogger = require('knex-tiny-logger').default;
const config = require('../../knexfile')[environment];

// eslint-disable-next-line import/order
const db = require('knex')(config);

const { DEBUG } = process.env;
if (DEBUG) knexTinyLogger(db);

module.exports = db;
