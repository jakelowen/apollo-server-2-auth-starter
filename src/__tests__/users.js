/* eslint-disable no-undef */
import gql from 'graphql-tag';

import { createTestClient } from 'apollo-server-testing';
import { constructTestServer, makeTestContext } from './__utils';
import db from '../db';
// import makeContext from '../utils/makeContext';
// import { makeTestContext } from './__utils';
// const nock = require('nock');

const ALL_USERS = gql`
    {
        users {
            items {
                id
                email
            }
            pageInfo {
                fromCache
                totalCount
                nextCursor
            }
        }
    }
`;

beforeAll(async () => db.migrate.latest());
beforeEach(async () => Promise.all([db.raw('TRUNCATE TABLE users CASCADE')]));
afterAll(async () => db.destroy());

describe('Queries', () => {
  test('fetches list of users if admin', async () => {
    await db.table('users').insert([
      { id: 'b6bf639f-6e80-4c6b-a46b-dc1373a395be', email: 'fake@test.com', name: 'Fake1' },
      { id: 'a6bf639f-6e80-4c6b-a46b-dc1373a395bf', email: 'fake2@test.com', name: 'Fake2' },
    ]);
    const { server } = constructTestServer({
      context: await makeTestContext(null, { user: { id: 1, email: 'a@a.a', roles: ['admin'] } }),
    });

    const { query } = createTestClient(server);
    const res = await query({ query: ALL_USERS });
    expect(res).toMatchSnapshot();
  });

  test('fetches list of users fails if not admin', async () => {
    await db.table('users').insert([
      { id: 'b6bf639f-6e80-4c6b-a46b-dc1373a395be', email: 'fake@test.com', name: 'Fake1' },
      { id: 'a6bf639f-6e80-4c6b-a46b-dc1373a395bf', email: 'fake2@test.com', name: 'Fake2' },
    ]);
    const { server } = constructTestServer({
      context: await makeTestContext(null, { user: { id: 1, email: 'a@a.a', roles: ['applicant'] } }),
    });

    const { query } = createTestClient(server);
    const res = await query({ query: ALL_USERS });
    expect(res).toMatchSnapshot();
  });
});
