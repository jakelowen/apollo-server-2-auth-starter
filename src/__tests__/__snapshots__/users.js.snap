// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches list of users fails if not admin 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: You must be an admin to perform this action.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches list of users if admin 1`] = `
Object {
  "data": Object {
    "users": Object {
      "items": Array [
        Object {
          "email": "fake2@test.com",
          "id": "a6bf639f-6e80-4c6b-a46b-dc1373a395bf",
        },
        Object {
          "email": "fake@test.com",
          "id": "b6bf639f-6e80-4c6b-a46b-dc1373a395be",
        },
      ],
      "pageInfo": Object {
        "fromCache": false,
        "nextCursor": null,
        "totalCount": 2,
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
